src/bitcoin-chainstate.cpp

```diff
@@ -102,6 +104,18 @@ int main(int argc, char* argv[])
         {
             std::cout << "Warning: " << warning.original << std::endl;
         }
+
+        bool fatalError(const std::string& strMessage, bilingual_str user_message) override
+        {
+            SetMiscWarning(Untranslated(strMessage));
+            LogPrintf("*** %s\n", strMessage);
+            if (user_message.empty()) {
+                user_message = _("A fatal internal error occurred, see debug.log for details");
+            }
+            std::cout << "Warning: " << user_message.original << std::endl;
+            StartShutdown();
+            return false;
```

Comment:
In commit "refactor: Pass FatalError in options to ChainstateManager" (2f9c2d245360b3fad6d469a76c2916d75b027b57)

I think this method should just be printing the messages to cout, letting error handling code below clean up after the failure. The chainstate example program shouldn't need to be logging things and calling StartShutdown.